synapse/handlers/saml_handler.py

```diff
@@ -173,22 +197,42 @@ def handle_redirect_request(self, client_redirect_url):
                     )
                     return registered_user_id
 
-            # figure out a new mxid for this user
-            base_mxid_localpart = self._mxid_mapper(mxid_source)
+            # Map saml response to user attributes using the configured mapping provider
+            for i in range(1000):
+                try:
+                    attribute_dict = self._user_mapping_provider.saml_response_to_user_attributes(
+                        saml2_auth, i
+                    )
+                except Exception:
+                    logging.exception("Error in SAML mapping provider plugin")
```

Comment:
don't do `logger.exception` *and* re-raise: it will log the stacktrace twice.

personally I'd remove the try/except here. I don't think there's any value in the extra log entry.